File: backends/python-fastapi/app/domain/staff/entity.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class Staff:
    """スタッフのドメインエンティティ（SQLAlchemy タグなし）。"""
    id: int = 0
    name: str = ""
    email: str = ""
    provider: int = 0
    provider_id: str = ""
    avatar: Optional[str] = None
    role: int = 0
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    deleted_at: Optional[datetime] = None

File: backends/rust-axum/src/domain/staff/entity.rs
//! スタッフドメイン エンティティモジュール。
//!
//! # Author
//! Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

use chrono::{DateTime, Utc};

/// スタッフのドメインエンティティ。
pub struct Staff {
    pub id:            u32,
    pub name:          String,
    pub email:         String,
    pub provider:      i32,
    pub provider_id:   String,
    pub avatar:        Option<String>,
    pub role:          i32,
    pub last_login_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub created_by:    Option<u32>,
    pub updated_at: DateTime<Utc>,
    pub updated_by:    Option<u32>,
    pub deleted_at: Option<DateTime<Utc>>,
    pub deleted_by:    Option<u32>,
    pub version:       i32,
}
